### PR TITLE
KOGITO-3497: Fail to load Decision model running dmn-tracing-quarkus example in Native mode

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/KOGITO-3497

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DefaultDecisionModelResource.java [hunk 1]
```diff
@@ -75,17 +68,12 @@ public DecisionModelType getModelType() {
...
-            LOG.error(ioe.getMessage());
-            throw new RuntimeException(ioe);
-        }
+        return load();
     }
 
-    private String load(InputStream is) {
+    private String load() {
         StringBuilder sb = new StringBuilder();
-        try (InputStreamReader isr = new java.io.InputStreamReader(is, StandardCharsets.UTF_8);
```
Comment (manstis) [inline drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DefaultDecisionModelResource.java:88]: This is the cause of the problem.

I've moved instantiation of the `InputStreamReader` to the generated code, as is performed for `DecisionModels`. A similar patch for the handling `InputStream` in GraalVM native for `DecisionModesl` was fixed by @evacchi for KOGITO-3330.

**Comment by manstis** [comment]
Jenkins retest this

**Review by Sgitario — APPROVED** [review]
Thanks! I've confirmed that these changes fixed the issue when running Kogito app using drools in quarkus native mode.

**Review by danielezonca — APPROVED** [review]
LGTM

**Pull request merged**